Plugins/Public/contraband/Main.cpp

```diff
@@ -0,0 +1,371 @@
+				{
+					string currentHouseCompare = iter->first;
+					list<uint> cfgCargoLst = iter->second;
+
+					if (currentHouse == currentHouseCompare)
+					{
+						for (list<CARGO_INFO>::iterator i = lstCargo.begin(); i != lstCargo.end(); ++i)
+						{
+							uint correctCargo = i->iArchID;
+							foreach(cfgCargoLst, uint, cfgCargoIter)
+							{
+								if (*cfgCargoIter == correctCargo) // If their cargo is on the list
+								{
+									ContrabandWarning(iClientID, L"WARNING: You are carrying items which are illegal in this sector.");
+									return;
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+//////////////////////////////////////////////
+//                 Commands                 //
+//////////////////////////////////////////////
+bool UserCmd_ManualCargoCheck(uint iClientID, const wstring & wscCmd, const wstring & wscParam, const wchar_t * usage)
+{
+	CheckCargo(iClientID);
+	return true;
+}
+
+bool UserCmd_Laws(uint iClientID, const wstring & wscCmd, const wstring & wscParam, const wchar_t * usage)
+{
+	if (set_lawsEnabled)
+	{
+		for (map<uint, string>::iterator iter = mapHouseAndSystems.begin(); iter != mapHouseAndSystems.end(); iter++)
+		{
+			uint currentSystem = iter->first;
+			string currentHouse = iter->second;
+
+			uint getSystem;
+			pub::Player::GetSystem(iClientID, getSystem);
+
+			if (currentSystem == getSystem)
+			{
+				for (map<string, wstring>::iterator iter2 = mapHouseLawList.begin(); iter2 != mapHouseLawList.end(); iter2++)
+				{
+					string currentHouseCompare = iter2->first;
+					if (currentHouse == currentHouseCompare)
+					{
+						wstring scXML = iter2->second;
+
+						wstring wscPlayerInfo = L"<RDL><PUSH/>" + scXML + L"<PARA/><POP/></RDL>";
+
+						struct PlayerData *pPD = 0;
+						while (pPD = Players.traverse_active(pPD))
+						{
+							uint iClientID = HkGetClientIdFromPD(pPD);
+
+							HkChangeIDSString(iClientID, 66076, wscPlayerInfo);
+							HkChangeIDSString(iClientID, 1, L"The Local Laws");
+
+							FmtStr caption(0, 0);
+							caption.begin_mad_lib(1);
+							caption.end_mad_lib();
+
+							FmtStr message(0, 0);
+							message.begin_mad_lib(66076);
+							message.end_mad_lib();
+
+							pub::Player::PopUpDialog(iClientID, caption, message, POPUPDIALOG_BUTTONS_CENTER_OK);
+						}
+					}
+				}
+
+			}
+		}
+		return false;
+	}
+	return true;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//Calls
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void __stdcall PlayerLaunch_AFTER(unsigned int iShip, unsigned int client)
+{
+	CheckCargo(client);
+}
+
+void __stdcall JumpInComplete(unsigned int system, unsigned int ship)
+{
+	uint iClientID = HkGetClientIDByShip(ship);
+	CheckCargo(iClientID);
+}
+
+/** Clean up when a client disconnects */
+void ClearClientInfo(uint iClientID)
```

Comment:
no need for this, we're not storing client-specific data

Reply:
HkChangeIDSString only changes it for a specific client, would this not warrant its use?

Reply:
No, it wouldn't. And if it did, this function here doesn't do anything so there'd be a problem anyway.